csrc/quantization/cutlass_w8a8/scaled_mm_entry.cu

```diff
@@ -120,9 +125,12 @@ void cutlass_scaled_mm(torch::Tensor& c, torch::Tensor const& a,
 
   // Guard against compilation issues for sm90 kernels
 #if defined ENABLE_SCALED_MM_C3X && ENABLE_SCALED_MM_C3X
-  if (version_num >= 90) {
+  if (version_num >= 90 && version_num < 100) {
     cutlass_scaled_mm_sm90(c, a, b, a_scales, b_scales, bias);
     return;
+  } else if (version_num >= 100) {
+    cutlass_scaled_mm_sm100(c, a, b, a_scales, b_scales, bias);
+    return;
   }
```

Comment:
This is causing linker errors in the CI. Need to guard against calling `cutlass_scaled_mm_sm100` when CUDA < 12.8

Something like this:
```
#if defined ENABLE_SCALED_MM_C3X && ENABLE_SCALED_MM_C3X

#if defined CUDA_VERSION && CUDA_VERSION >= 12800
  if (version_num >= 90) {
    cutlass_scaled_mm_sm90(c, a, b, a_scales, b_scales, bias);
    return;
  }
#else
  if (version_num >= 90 && version_num < 100) {
    cutlass_scaled_mm_sm90(c, a, b, a_scales, b_scales, bias);
    return;
  } else if (version_num >= 100) {
    cutlass_scaled_mm_sm100(c, a, b, a_scales, b_scales, bias);
    return;
  }
#endif
```